dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/StatsTaskListGenerator.java

```diff
@@ -70,21 +76,37 @@ ImmutableList<Task<?>> createTasks(ConnectorArguments arguments, Duration querie
     }
     for (String name : nativeNames(/* required= */ true)) {
       OracleStatsQuery item =
-          OracleStatsQuery.createNative(name, /* isRequired= */ true, queriedDuration);
+          OracleStatsQuery.createNative(
+              name, /* isRequired= */ true, queriedDuration, MULTI_TENANT);
       builder.add(StatsJdbcTask.fromQuery(item));
     }
     for (String name : nativeNames(/* required= */ false)) {
       OracleStatsQuery item =
-          OracleStatsQuery.createNative(name, /* isRequired= */ false, queriedDuration);
+          OracleStatsQuery.createNative(
+              name, /* isRequired= */ false, queriedDuration, MULTI_TENANT);
       builder.add(StatsJdbcTask.fromQuery(item));
     }
     for (String name : statspackNames()) {
       OracleStatsQuery query = OracleStatsQuery.createStatspack(name, queriedDuration);
       builder.add(StatsJdbcTask.fromQuery(query));
     }
+    for (String name : NATIVE_NAMES_OPTIONAL) {
+      builder.addAll(createTaskWithAlternative(name, false, queriedDuration));
+    }
     return builder.build();
   }
 
+  List<Task<?>> createTaskWithAlternative(
+      String name, boolean isRequired, Duration queriedDuration) {
+    OracleStatsQuery primary =
+        OracleStatsQuery.createNative(name, /* isRequired= */ false, queriedDuration, MULTI_TENANT);
+    StatsJdbcTask primaryTask = StatsJdbcTask.fromQuery(primary);
+    OracleStatsQuery alternative =
+        OracleStatsQuery.createNative(name, isRequired, queriedDuration, SINGLE_TENANT);
```

Comment:
Short answer: it will succeed.

But, this is a very good question, I think it would be best to discuss it in detail when there's time. A "required" task is not actually required, all that this enum does is decide the error handling strategy:
- `required` - propagate any errors
- `optional` - log errors and continue
- `informational` - same as `optional`

Relevant code: https://github.com/google/dwh-migration-tools/blob/eaeadef2449b465b59633f2034ea55104e70bf01/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java#L256

In a future fix, maybe we could either change the behavior to actually require the `required` tasks or change the names to not suggest that this is the case.